jax/_src/dtypes.py

```diff
@@ -249,6 +249,7 @@ def issubdtype(a, b):
 issubsctype = np.issubsctype
 
 # Enumeration of all valid JAX types in order.
```

Comment:
In what order?? (Not this change, so feel free to ignore)

Reply:
I think the comment was meant to reflect that some of the code below depends on the order of these enumerations.

Reply:
I expanded the comment and moved the new list above to make this more clear.